proto/asn1-pdu/asn1_universal_types_to_der.cc

```diff
@@ -23,25 +23,38 @@
 
 namespace asn1_universal_types {
 
+void Encode(const Boolean& boolean, std::vector<uint8_t>& der) {
+  der.push_back(kAsn1Boolean);
+  // The contents octets shall consist of a single octet (X.690 (2015), 8.2.1).
+  // Therefore, length is always 1.
+  der.push_back(0x01);
+
+  if (boolean.val()) {
+    // If the boolean value is TRUE, the octet shall have any non-zero value
+    // (X.690 (2015), 8.2.2).
+    der.push_back(0xFF);
+  } else {
+    // If the boolean value is FALSE, the octet shall be zero (X.690
+    // (2015), 8.2.2).
+    der.push_back(0x00);
+  }
+}
+
 void Encode(const Integer& integer, std::vector<uint8_t>& der) {
-  // Save the current size in |tag_len_pos| to place tag and length
-  // after the value is encoded.
-  const size_t tag_len_pos = der.size();
+  EncodeTagAndLength(kAsn1Integer, der.size(),
+                     std::min<uint8_t>(0x01, integer.val().size()), der);
```

Comment:
This isn't a uint8_t though :) I tried to fix it in the suggestion, but apologies for not explicitly calling it out.

It should automatically promote the 0x01 to the size of the larger (unsigned) integer, integer.val.size(). If that's not happening,  you may just be missing a `0x01u`, although if it's not promoting, you probably want to std::min<size_t> to pick the larger type, not the smaller type :)

Reply:
Sorry I think I'm a bit slower this morning. This should have been obvious in your last comment.